### PR TITLE
[Qt] Ensure that mouse position for wxMouseEvent is relative to the window raising it

### PR DESCRIPTION
**Pull request by matthew-griffin**
Discovered that mouse position received in wxMouseEvents was incorrect if the event was raised by a sub-widget within qt wrappers (as is the case for wxListCtrl). The mouse position was relative to the sub-widget that wx has no knowledge of. We can instead convert the original screen position to the client space of the current window when creating the wx event.

### CODE REVIEW

File: src/qt/window.cpp [hunk 1]
```diff
@@ -1387,12 +1388,19 @@ bool wxWindowQt::QtHandleMouseEvent ( QWidget *handler, QMouseEvent *event )
             wxFAIL_MSG( "Unknown mouse event type" );
     }
 
-    // Fill the event
-    QPoint mousePos = event->pos();
+    // Fill the event - convert screen position to the window's coordinates
+    wxPoint mousePos = ScreenToClient(wxQtConvertPoint(event->globalPos()));
+
+    // Correct the mouse position if we have a caption
+    if (HasFlag(wxCAPTION))
```
Comment (catalinr) [inline src/qt/window.cpp]: Hmm, will this still be needed after #1402 will have been applied?

Comment (matthew-griffin) [reply src/qt/window.cpp]: I assume so as most of our changes have been merged into our [gmd_consolidation_branch](https://github.com/GeoTeric/wxWidgets/commits/gmd_consolidation_branch), which didn't cause a conflict and must have been an additional requirement.

Comment (catalinr) [reply src/qt/window.cpp]: Can you please check? It does sound like it should not be needed once the client area is correctly calculated (or maybe `ScreenToClient()` still has a bug in it?)

Comment (matthew-griffin) [reply src/qt/window.cpp]: I've tried removing this code in our consolidation branch and found that it is still necessary to be able to un-dock aui panes, like the Tree Pane in the 'auidemo' sample. I believe this also caused an issue with being able to click on the close button of aui panes if they were initialised to an un-docked state.

Comment (catalinr) [reply src/qt/window.cpp]: I don't really understand the reference to AUI here (was that comment intended for this PR?), but to rephrase the question: are the change in #1402 part of your _consolidation branch_?

Comment (matthew-griffin) [reply src/qt/window.cpp]: Yes, apologies if it was unclear but both this change and #1402 are part of our consolidation branch.

The AUI comment was what happens if I try removing the `if (HasFlag(wxCAPTION))` section combined with #1402.

Comment (matthew-griffin) [reply src/qt/window.cpp]: Yes, apologies if it was unclear but both this change and #1402 are part of our consolidation branch.

The AUI comment was what happens if I try removing the 'if (HasFlag(wxCAPTION))' section combined with #1402.

Comment (catalinr) [reply src/qt/window.cpp]: But then, if there is a problem with AUI, shouldn't it be dealt with in AUI-specific code?
My concern here is about native TLWs, i.e. `wxFrame` and `wxDialog` - what happens for them? Is that "correction" required or not for them?

Comment (matthew-griffin) [reply src/qt/window.cpp]: I think this change was added here as the AUI code is largely generic and not split for different versions. I personally don't know of any other usages of `wxCAPTION` where this adjustment may or may not be necessary but it's certainly not present for all TLWs. Up to you whether you think it should come in I guess.

Comment (catalinr) [reply src/qt/window.cpp]: > I personally don't know of any other usages of `wxCAPTION` where this adjustment may or may not be necessary but it's certainly not present for all TLWs. Up to you whether you think it should come in I guess.

Ok, there are 2 things that do not add up:
1. `wxCAPTION` is not used by AUI, is it? So how could the check for this symbol fix an AUI problem?
2. It's not up to me or anyone else, it's up to whether it works better with that check or not. In conclusion, that code should be tested with a `wxFrame`, with and without caption, and confirm that mouse events get the correct position, as opposed to the code not containing that condition (and if it's still the case, maybe explain in the process, why such a condition is still necessary).

Comment (matthew-griffin) [reply src/qt/window.cpp]: Apologies as my colleague added the `wxCAPTION` section so I'm not 100% sure, but I assumed that AUI did make use of a caption for its title bar, which was not being taken into account and was the reason the AUI Panes could not be dragged by their title bar to dock and undock them.

I'm afraid I don't have the time to run additional tests on frames with and without captions, I can only say that we haven't seen any problems with frames that don't have captions elsewhere. That's why I would suggest potentially removing the wxCAPTION section if we can't confirm all scenarios have the correct position right now.

**Comment by vadz** [comment]
Sorry, I don't think the test for `wxCAPTION` makes sense here. It probably just compensates for the fact that `ScreenToClient()` doesn't work correctly for TLWs in wxQt.

I'll apply the rest of the patch, but please fix `ScreenToClient()` instead, if needed. TIA!

**Pull request closed without merging**